DeusEx/Classes/Player.uc

```diff
@@ -9,3 +9,144 @@ function bool AddInventory( inventory NewItem )
+{
+    local float newDamage, oldDamage;
+    local float augLevel, skillLevel;
+    local float pct;
+    local HazMatSuit suit;
+    local BallisticArmor armor;
+    local bool bReduced;
+
+    bReduced = False;
+    newDamage = Float(Damage);
+    oldDamage = newDamage;
+
+    if ((damageType == 'TearGas') || (damageType == 'PoisonGas') || (damageType == 'Radiation') ||
+        (damageType == 'HalonGas')  || (damageType == 'PoisonEffect') || (damageType == 'Poison') 
+        /*|| damageType == 'Flamed' || damageType == 'Burned'*/ )
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugEnviro');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+
+        // get rid of poison if we're maxed out
+        if (newDamage ~= 0.0)
+        {
+            StopPoison();
+            drugEffectTimer -= 4;	// stop the drunk effect
+            if (drugEffectTimer < 0)
+                drugEffectTimer = 0;
+        }
+
+        if (UsingChargedPickup(class'HazMatSuit'))
+        {
+            skillLevel = SkillSystem.GetSkillLevelValue(class'SkillEnviro');
+            newDamage *= 0.75 * skillLevel;
+        }
+    }
+
+    if ((damageType == 'Shot') || (damageType == 'Sabot') || (damageType == 'Exploded') || (damageType == 'AutoShot'))
+    {
+        // go through the actor list looking for owned BallisticArmor
+        // since they aren't in the inventory anymore after they are used
+        if (UsingChargedPickup(class'BallisticArmor'))
+        {
+            skillLevel = SkillSystem.GetSkillLevelValue(class'SkillEnviro');
+            newDamage *= 0.5 * skillLevel;
+        }
+    }
+
+    if (damageType == 'HalonGas')
+    {
+        if (bOnFire && !bCheckOnly)
+            ExtinguishFire();
+    }
+
+    if ((damageType == 'Shot') || (damageType == 'AutoShot'))
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugBallistic');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+    }
+
+    if (damageType == 'EMP')
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugEMP');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+    }
+
+    if ((damageType == 'Burned') || (damageType == 'Flamed') ||
+        (damageType == 'Exploded') || (damageType == 'Shocked'))
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugShield');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+    }
+
+    //
+    // Reduce or increase the damage based on the combat difficulty setting, do this before SetDamagePercent for the UI display
+    // because we don't want to show 100% damage reduction but then do the minimum of 1 damage
```

Comment:
we do the double scaling of combatDifficulty so that the rounding matches up

compare to the original code, which did it kinda backwards, and with potential rounding errors in the UI showing damage reduction
```
	if (newDamage < Damage)
	{
		if (!bCheckOnly)
		{
			pct = 1.0 - (newDamage / Float(Damage));
			SetDamagePercent(pct);
			ClientFlash(0.01, vect(0, 0, 50));
		}
		bReduced = True;
	}
	else
	{
		if (!bCheckOnly)
			SetDamagePercent(0.0);
	}


	//
	// Reduce or increase the damage based on the combat difficulty setting
	//
	if ((damageType == 'Shot') || (damageType == 'AutoShot'))
	{
		newDamage *= CombatDifficulty;

		// always take at least one point of damage
		if ((newDamage <= 1) && (Damage > 0))
			newDamage = 1;
	}
```

Reply:
I'm actually not sure if the damage reduction % taking rounding into account is good or bad, might just make it more complicated for the player to understand